### PR TITLE
Allow heartbeat to be disabled

### PR DESCRIPTION
**Pull request by kevtaylor**

### CODE REVIEW

File: charts/linkerd2/values.yaml [hunk 1]
```diff
@@ -47,6 +47,7 @@ Identity:
 GrafanaImage: gcr.io/linkerd-io/grafana
 
 # heartbeat configuration
+Heartbeat: false
```
[the remark this answers is not in the document]
Comment (kevtaylor) [reply charts/linkerd2/values.yaml]: I dont quite get your comment. Are you saying that you want Heartbeat enabled by default and end user switches it off? So DisableHeartBeat:: true

[the remark this answers is not in the document]
Comment (kevtaylor) [reply charts/linkerd2/values.yaml]: With default set as DisableHeartBeat: false

[the remark this answers is not in the document]
Comment (grampelberg) [reply charts/linkerd2/values.yaml]: Yup! That was my thinking, similar to `RestrictDashboardPrivileges`.

[the remark this answers is not in the document]
Comment (kevtaylor) [reply charts/linkerd2/values.yaml]: When attempting to get the tests running, the go mod fails 
```
go: contrib.go.opencensus.io/exporter/ocagent@v0.2.0: unrecognized import path "contrib.go.opencensus.io/exporter/ocagent" (https fetch: Get https://contrib.go.opencensus.io/exporter/ocagent?go-get=1: read tcp 10.0.2.15:50910->216.239.36.21:443: read: connection reset by peer)
```

[the remark this answers is not in the document]
Comment (kevtaylor) [reply charts/linkerd2/values.yaml]: And also
```
go: gopkg.in/inf.v0@v0.9.1: unrecognized import path "gopkg.in/inf.v0" (https fetch: Get https://gopkg.in/inf.v0?go-get=1: dial tcp 35.196.143.184:443: i/o timeout)
```

[the remark this answers is not in the document]
Comment (kevtaylor) [reply charts/linkerd2/values.yaml]: I deleted the go.sum file and let it rebuild it. That is checked in in this commit. The build and tests all passed locally, but I can revert that sum file if needed. I addressed the other comments also

[the remark this answers is not in the document]
Comment (kevtaylor) [reply charts/linkerd2/values.yaml]: Okay, forget all that, I see I need to rebase.

[the remark this answers is not in the document]
Comment (kevtaylor) [reply charts/linkerd2/values.yaml]: Not sure what's happening now, as there seems to be a whole lot of new stuff in the build process which is failing.

[the remark this answers is not in the document]
Comment (siggy) [reply charts/linkerd2/values.yaml]: Sorry about that, you caught us in a CI migration. If you merge master now your tests should pass.

**Comment by ihcsim** [comment]
@kevtaylor Thanks for the PR! I've done some testing on your feature branch, using both `helm install --set DisableHeartBeat=true` and `linkerd install`. The installation works as intended :+1:, but with the `heartbeat` service disabled, the `linkerd check` command failed with:

```
× control plane ServiceAccounts exist
    missing ServiceAccounts: linkerd-heartbeat
    see https://linkerd.io/checks/#l5d-existence-sa for hints
```

The changes required to fix this is a bit more involved. After chatting with @grampelberg, we agree to create https://github.com/linkerd/linkerd2/issues/3386 to track the task to fix `linkerd check`.

So once your merged your branch with the latest `master` to fix the CI failure, we can merge this branch. But keep in mind that, with `DisableHeartBeat: true`, the `linkerd check` command will fai until https://github.com/linkerd/linkerd2/issues/3386 is merged. (If you are interested in tackling https://github.com/linkerd/linkerd2/issues/3386, I can provide you with pointers on where the code is.)

**Comment by grampelberg** [comment]
@kevtaylor we love sending contributors some swag. Mind filling out the [form](https://docs.google.com/forms/d/e/1FAIpQLSdXA7kAl6oyUQ4DFRjiDc2biWlvkhNirrMwsXaQ8V3Uh8Yj5A/viewform?usp=sf_link) so that we can get that to you? =)

**Pull request merged**